api/auth/ldap/ldap.go

```diff
@@ -81,3 +116,46 @@ func WithMountPath(mountPath string) LoginOption {
 		return nil
 	}
 }
+
+func (a *LDAPAuth) readPasswordFromFile() (string, error) {
+	passwordFile, err := os.Open(a.passwordFile)
+	if err != nil {
+		return "", fmt.Errorf("unable to open file containing password: %w", err)
+	}
+	defer passwordFile.Close()
+
+	limitedReader := io.LimitReader(passwordFile, 1000)
+	passwordBytes, err := io.ReadAll(limitedReader)
+	if err != nil {
+		return "", fmt.Errorf("unable to read password: %w", err)
+	}
+
+	passwordValue := strings.TrimSuffix(string(passwordBytes), "\n")
+
+	return passwordValue, nil
+}
+
+func (password *Password) validate() error {
+	if password.FromFile == "" && password.FromEnv == "" && password.FromString == "" {
+		return fmt.Errorf("password for Userpass auth must be provided with a source file, environment variable, or plaintext string")
```

Comment:
Appreciate it! Just one more Userpass -> LDAP change that needs to happen here and then we’re good.

Reply:
All set!